pkgs/development/python-modules/tree-sitter-grammars/default.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, buildPythonPackage
+, pytestCheckHook
+, tree-sitter
+, symlinkJoin
+, writeTextDir
+  # `name`: grammar derivation pname in the format of `tree-sitter-<lang>`
+, name
+, grammarDrv
+}:
+let
+  inherit (grammarDrv) version;
+
+  snakeCaseName = lib.replaceStrings [ "-" ] [ "_" ] name;
+  drvPrefix = "python-${name}";
+  langIdentOverrides = {
+    tree_sitter_org_nvim = "tree_sitter_org";
+  };
+  langIdent = langIdentOverrides.${snakeCaseName} or snakeCaseName;
+in
+assert (lib.assertMsg (!(grammarDrv.meta ? license)) ''
+  As of this writing, ${grammarDrv.pname} surprisingly doesn't have a license.
+  This trap is set here to guarantee that if it ever does have a license, this package will inherit the license.
+'');
+buildPythonPackage {
+  inherit version;
+  pname = drvPrefix;
+
+  src = symlinkJoin {
+    name = "${drvPrefix}-source";
+    paths = [
+      (writeTextDir "${snakeCaseName}/__init__.py"
+        ''
+          from ._binding import language
+
+          __all__ = ["language"]
+        ''
+      )
+      (writeTextDir "${snakeCaseName}/binding.c"
+        ''
+          #include <Python.h>
+
+          typedef struct TSLanguage TSLanguage;
+
+          TSLanguage *${langIdent}(void);
+
+          static PyObject* _binding_language(PyObject *self, PyObject *args) {
+              return PyLong_FromVoidPtr(${langIdent}());
+          }
+
+          static PyMethodDef methods[] = {
+              {"language", _binding_language, METH_NOARGS,
+              "Get the tree-sitter language for this grammar."},
+              {NULL, NULL, 0, NULL}
+          };
+
+          static struct PyModuleDef module = {
+              .m_base = PyModuleDef_HEAD_INIT,
+              .m_name = "_binding",
+              .m_doc = NULL,
+              .m_size = -1,
+              .m_methods = methods
+          };
+
+          PyMODINIT_FUNC PyInit__binding(void) {
+              return PyModule_Create(&module);
+          }
+        ''
+      )
+      (writeTextDir "setup.py"
+        ''
+          from platform import system
+          from setuptools import Extension, setup
+
+
+          setup(
```

Comment:
Most of this should be moved to `pyproject.toml`, see https://setuptools.pypa.io/en/latest/userguide/ext_modules.html#building-extension-modules

Reply:
Hi, thanks for the suggestion. Indeed, the file is present upstream and I added it.

However, since everything is generated, I am not sure if it is really helpful for us now?